shell/browser/web_contents_preferences.cc

```diff
@@ -459,24 +460,19 @@ void WebContentsPreferences::OverrideWebkitPrefs(
 
   // Run Electron APIs and preload script in isolated world
   prefs->context_isolation = context_isolation_;
-  prefs->guest_instance_id = guest_instance_id_;
+  prefs->is_webview = is_webview_;
 
   prefs->hidden_page = false;
-  if (guest_instance_id_) {
-    // Webview `document.visibilityState` tracks window visibility so we need
-    // to let it know if the window happens to be hidden right now.
-    auto* manager = WebViewManager::GetWebViewManager(web_contents_);
-    if (manager) {
-      auto* embedder = manager->GetEmbedder(guest_instance_id_);
-      if (embedder) {
-        auto* relay = NativeWindowRelay::FromWebContents(embedder);
-        if (relay) {
-          auto* window = relay->GetNativeWindow();
-          if (window) {
-            const bool visible = window->IsVisible() && !window->IsMinimized();
-            if (!visible) {
-              prefs->hidden_page = true;
-            }
+  // Webview `document.visibilityState` tracks window visibility so we need
+  // to let it know if the window happens to be hidden right now.
+  if (auto* api_web_contents = api::WebContents::From(web_contents_)) {
+    if (auto* embedder = api_web_contents->embedder()) {
```

Comment:
is it guaranteed that this is always the same WebContents as would previously have been returned by looking up through `guest_instance_id_`?

Reply:
according to the code, that should be the case